### PR TITLE
Patch for Issue 1203

### LINKED ISSUE #1203: Can't unparse `#KRewrite`
The `kast` unparser can't deal with the `#KRewrite` symbol in Kore. Any way you put a Kore rule (with a rewrite) through the unparser should give the same result, but here is a small example using the `kore-repl`

Files: `foo.k`, `foo-spec.k`, `kast.kscript`

`foo.k`:

``` 
module FOO                                                                                                                                                                                                                                                       
     imports DOMAINS                                                              

     configuration                                                                
         <k> $PGM:Val </k>                                                        

     syntax Val ::= "foo" | "bar"                                                 
     rule <k> foo => bar ... </k>                                                 
     rule <k> bar => 0   ... </k>                                                                                                                                  
 endmodule
```

`foo-spec.k`:

```
module FOO-SPEC
     import FOO

     rule <k> foo => 0 ... </k>
 endmodule
```

`kast.kscript`
```
alias krule = rule | kast --debug -i kore -o pretty -d . /dev/stdin
```

Commands:
```
kompile --backend haskell foo.k
kprove --haskell-backend-command "$YOUR_K_REPO/k/k-distribution/target/release/k/bin/kore-repl --repl-script ./kast.kscript" foo-spec.k -m FOO
```

In the repl:
```
Kore 0> step
Kore 1> krule
```

Expected result: Prints the rule `<k> foo => bar ... </k>`
Actual result:
```
java.util.NoSuchElementException: key not found: #KRewrite
        at scala.collection.MapLike.default(MapLike.scala:232)
        at scala.collection.MapLike.default$(MapLike.scala:231)
        at scala.collection.AbstractMap.default(Map.scala:59)
        at scala.collection.MapLike.apply(MapLike.scala:141)
        at scala.collection.MapLike.apply$(MapLike.scala:140)
        at scala.collection.AbstractMap.apply(Map.scala:59)
        at org.kframework.unparser.KOREToTreeNodes$.apply(KOREToTreeNodes.scala:47)
        at org.kframework.unparser.KOREToTreeNodes.apply(KOREToTreeNodes.scala)
        at org.kframework.unparser.KPrint.unparseInternal(KPrint.java:192)
        at org.kframework.unparser.KPrint.unparseTerm(KPrint.java:186)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:139)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:125)
        at org.kframework.unparser.KPrint.prettyPrint(KPrint.java:121)
        at org.kframework.kast.KastFrontEnd.run(KastFrontEnd.java:137)
        at org.kframework.main.FrontEnd.main(FrontEnd.java:59)
        at org.kframework.main.Main.runApplication(Main.java:119)
        at org.kframework.main.Main.runApplication(Main.java:108)
        at org.kframework.main.Main.main(Main.java:54)
[Error] Internal: Uncaught exception thrown of type NoSuchElementException (key
not found: #KRewrite)
/home/hjorthjort/tmp/k-errors/rewrite-unparse/./foo.k:8:10-8:33
Axiom 0
```

### PR DESCRIPTION
**Pull request by traiansf**

### CODE REVIEW

File: kernel/src/main/java/org/kframework/kast/KastFrontEnd.java [hunk 1]
```diff
@@ -112,7 +116,10 @@ public int run() {
                 options.module = def.mainSyntaxModuleName();
                 switch (options.input) {
                     case KORE:
-                        unparsingMod = def.languageParsingModule();
+                        Module languageParsingModule = def.languageParsingModule();
+                        Module k = def.kompiledDefinition.getModule("K").get();
+                        unparsingMod = Constructors.Module("K-LANGUAGE-PARSING",
+                                Set(languageParsingModule, k) , Set(), KORE.Att());
```
Comment (ttuegel) [inline kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: What's going on here?

Comment (dwightguth) [reply kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: Yeah, this seems strange.

Comment (traiansf) [reply kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: Well, the `#KRewrite` label is defined in the `K` module, which is used for parsing rules and not being computed for the compiled definition. 

I'm just creating a module importing both the k module and the parsing module here on-spot, to make `#KRewrite` available to the definition.

Do you have any suggestions?

Comment (dwightguth) [reply kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: Just create a new module in kast.k containing just the rewrite production and import it in both K and K-TERM.

Comment (traiansf) [reply kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: done. please take another look.

**Review by dwightguth — APPROVED** [review]

File: kernel/src/main/java/org/kframework/kast/KastFrontEnd.java [hunk 2]
```diff
@@ -5,9 +5,11 @@
 import com.google.inject.Provider;
 import org.kframework.attributes.Source;
 import org.kframework.compile.ExpandMacros;
+import org.kframework.definition.Constructors;
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/kast/KastFrontEnd.java]: looks like you can revert the changes to this file.

**Comment by traiansf** [comment]
@dwightguth : there seem to be some issues with this change; using krewrite as such triggers a timeout in the unparsing tests. any suggestions on how to better address this?

**Pull request merged**